File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.go
func maxProfit(prices []int) int {

	left := 0
	max := 0
	for r:=1; r < len(prices); r++{
		diference := prices[r] - prices[left]
		if diference < 0{
			left = r
		}
		if diference > max{
			max = diference
		}
	}
	return max
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        
        self.diameter = 0  # store the maximum diameter found

        def height(node):
            if not node:
                return 0
            left_height = height(node.left)
            right_height = height(node.right)

            # update diameter with path through this node
            self.diameter = max(self.diameter, left_height + right_height)

            # return height to parent
            return 1 + max(left_height, right_height)

        height(root)
        return self.diameter